src/NUnitEngine/nunit.engine/Services/DomainManager.cs

```diff
@@ -149,6 +152,38 @@ public void Unload(AppDomain domain)
             new DomainUnloader(domain).Unload();
         }
 
+        public static bool TryGetConfigFileName(string settingsConfigFile, string testFile, string appBaseDir, out string config)
+        {
+            if (!string.IsNullOrEmpty(settingsConfigFile))
+            {
+                config = settingsConfigFile;
+                return true;
+            }
+
+            if (string.IsNullOrEmpty(testFile))
+            {
+                config = null;
+                return false;
+            }
+
+            var configFile = testFile + ".config";
+            var configFileName = Path.GetFileName(configFile);
+            if (!StringComparer.InvariantCultureIgnoreCase.Equals(configFile, configFileName))
+            {
+                config = configFile;
+                return true;
+            }
+
+            if (!string.IsNullOrEmpty(appBaseDir))
+            {
+                config = Path.Combine(appBaseDir, Path.GetFileName(configFile));
```

Comment:
You don't need to `GetFileName` again, it is already in the variable `configFileName`.